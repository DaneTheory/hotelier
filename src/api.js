export const RUNNING = 'running';
export const STOPPED = 'stopped';
export const CRASHED = 'crashed';

export default class HotelApi {
  static getServers = () => {
    return window.fetch(`${HotelApi.getHotelUrl()}/_/servers`).then(response => {
      if (response.ok) {
        return response.json();
      } else {
        return [];
      }
    });
  };

  static watchServers = (cb) => {
    setInterval(() => HotelApi.getServers().then(data => cb(data)), 1000);
  }

  static getHotelUrl = () => {
    return process.env.NODE_ENV === 'development' ? '' : 'http://localhost:2000';
  }

  static sendCommand = (id, command) => {
    return HotelApi.postData(`${HotelApi.getHotelUrl()}/_/servers/${id}/${command}`, null);
  }

  static startServer = (id) => {
    return HotelApi.sendCommand(id, 'start');
  }

  static stopServer = (id) => {
    return HotelApi.sendCommand(id, 'stop');
  }

  static postData = (url, data) => {
    return fetch(url, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
      },
      body: JSON.stringify(data),
    });
  }
}
