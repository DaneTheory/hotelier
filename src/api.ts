export default class HotelApi {
  static getServers = () => {
    return window.fetch(`${HotelApi.getHotelUrl()}/_/servers`).then(response => {
      if (response.ok) {
        return response.json();
      } else {
        return [];
      }
    });
  }

  static watchServers = (cb: any) => {
    setInterval(() => HotelApi.getServers().then(data => cb(data)), 3000);
  }

  static getHotelUrl = () => {
    // In development, we have a reverse proxy to localhost:2000 (see package.json)
    return process.env.NODE_ENV === 'development' ? '' : window.hotelUrl;
  }

  static sendCommand = (id: string, command: string) => {
    return HotelApi.postData(`${HotelApi.getHotelUrl()}/_/servers/${id}/${command}`, null);
  }

  static startServer = (id: string) => {
    return HotelApi.sendCommand(id, 'start');
  }

  static stopServer = (id: string) => {
    return HotelApi.sendCommand(id, 'stop');
  }

  static postData = (url: string, data: any) => {
    return fetch(url, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
      },
      body: JSON.stringify(data),
    });
  }
}
