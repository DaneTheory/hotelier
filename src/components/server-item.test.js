import React from 'react';
import ServerItem from './server-item';
import { shallow } from 'enzyme';
import { STOPPED, RUNNING, RESTARTING } from '../api';

describe('ServerItem', () => {
  let runningServer;
  let stoppedServer;

  beforeEach(() => {
    runningServer = { id: 'test-server-1', status: RUNNING };
    stoppedServer = { id: 'test-server-2', status: STOPPED };
  });

  describe('#componentDidUpdate', () => {
    it('does not update the state if server is restarting', () => {
      const restartServer = jest.fn();
      const rendered = shallow(<ServerItem server={runningServer} restartServer={restartServer} />);
      rendered.instance().restartServer(runningServer);
      const newServerState = Object.assign({}, runningServer);
      newServerState.status = STOPPED;
      rendered.setProps({ server: newServerState });
      expect(rendered.instance().state.server.status).toEqual(RESTARTING);
    });

    it('updates the state from new props if server is not restarting', () => {
      const rendered = shallow(<ServerItem server={runningServer} />);
      const newServerState = Object.assign({}, runningServer);
      newServerState.status = STOPPED;
      rendered.setProps({ server: newServerState });
      expect(rendered.instance().state.server.status).toEqual(STOPPED);
    });
  });

  it('matches the snapshot when the server is running', () => {
    const rendered = shallow(<ServerItem server={runningServer} />);
    expect(rendered).toMatchSnapshot();
  });

  it('matches the snapshot when the server is stopped', () => {
    const rendered = shallow(<ServerItem server={stoppedServer} />);
    expect(rendered).toMatchSnapshot();
  });

  describe('#actionClassName', () => {
    let instance;

    beforeEach(() => instance = shallow(<ServerItem server={runningServer} />).instance());

    it('is positive when status is STOPPED', () => {
      expect(instance.actionClassName(STOPPED)).toEqual('tcon-transform btn-default');
    });

    it('is negative when status is RUNNING', () => {
      expect(instance.actionClassName(RUNNING)).toEqual('btn-negative text-white');
    });

    it('is neutral when status is unknown', () => {
      expect(instance.actionClassName('bla')).toEqual('tcon-transform btn-default');
    });
  });

  it('can open server by clicking on the name', () => {
    const openServer = jest.fn();
    const wrapper = shallow(<ServerItem server={runningServer} openServer={openServer} />);
    wrapper.find('.server-name').first().simulate('click');
    expect(openServer).toHaveBeenCalled();
  });

  it('can toggle server by clicking on the action button', () => {
    const toggleServer = jest.fn();
    const wrapper = shallow(<ServerItem server={runningServer} toggleServer={toggleServer} />);
    wrapper.find('button').last().simulate('click');
    expect(toggleServer).toHaveBeenCalled();
  });

  describe('restart button', () => {
    it('can restart server by clicking on the restart button', () => {
      jest.useFakeTimers();
      const restartServer = jest.fn();
      const wrapper = shallow(<ServerItem server={runningServer} restartServer={restartServer} />);
      wrapper.find('.restart-button').first().simulate('click');
      expect(restartServer).toHaveBeenCalled();
      expect(runningServer.status).toEqual(RESTARTING);
      expect(wrapper.instance().state.server.status).toEqual(RESTARTING);
      jest.runOnlyPendingTimers();
      expect(runningServer.status).toEqual(RUNNING);
      expect(wrapper.instance().state.server.status).toEqual(RUNNING);
    });
  });
});
