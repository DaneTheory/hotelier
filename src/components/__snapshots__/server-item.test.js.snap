// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerItem matches the snapshot when the server is running 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServerItem
    server={
        Object {
            "id": "test-server-1",
            "status": "running",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="pull-left server-name server-name--running"
          onClick={[Function]}
>
          test-server-1
</div>,
        <div
          className="pull-right"
>
          <div
                    className="d-inline"
          >
                    <CSSTransition
                              classNames="server-item"
                              in={true}
                              timeout={
                                        Object {
                                                  "enter": 350,
                                                  "exit": 350,
                                                }
                              }
                              unmountOnExit={true}
                    >
                              <button
                                        className="btn btn-default mr-1 restart-button"
                                        onClick={[Function]}
                              >
                                        <span
                                                  className="icon icon-arrows-ccw "
                                        />
                              </button>
                    </CSSTransition>
          </div>
          <div
                    className="d-inline"
          >
                    <button
                              className="btn btn-negative text-white"
                              onClick={[Function]}
                              type="button"
                    >
                              <span
                                        className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
                              />
                    </button>
          </div>
</div>,
      ],
      "className": "list-group-item server-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test-server-1",
          "className": "pull-left server-name server-name--running",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "test-server-1",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="d-inline"
>
              <CSSTransition
                            classNames="server-item"
                            in={true}
                            timeout={
                                          Object {
                                                        "enter": 350,
                                                        "exit": 350,
                                                      }
                            }
                            unmountOnExit={true}
              >
                            <button
                                          className="btn btn-default mr-1 restart-button"
                                          onClick={[Function]}
                            >
                                          <span
                                                        className="icon icon-arrows-ccw "
                                          />
                            </button>
              </CSSTransition>
</div>,
            <div
              className="d-inline"
>
              <button
                            className="btn btn-negative text-white"
                            onClick={[Function]}
                            type="button"
              >
                            <span
                                          className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
                            />
              </button>
</div>,
          ],
          "className": "pull-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CSSTransition
                classNames="server-item"
                in={true}
                timeout={
                                Object {
                                                "enter": 350,
                                                "exit": 350,
                                              }
                }
                unmountOnExit={true}
>
                <button
                                className="btn btn-default mr-1 restart-button"
                                onClick={[Function]}
                >
                                <span
                                                className="icon icon-arrows-ccw "
                                />
                </button>
</CSSTransition>,
              "className": "d-inline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn btn-default mr-1 restart-button"
                  onClick={[Function]}
>
                  <span
                                    className="icon icon-arrows-ccw "
                  />
</button>,
                "classNames": "server-item",
                "in": true,
                "timeout": Object {
                  "enter": 350,
                  "exit": 350,
                },
                "unmountOnExit": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="icon icon-arrows-ccw "
/>,
                  "className": "btn btn-default mr-1 restart-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon icon-arrows-ccw ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-negative text-white"
                onClick={[Function]}
                type="button"
>
                <span
                                className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
                />
</button>,
              "className": "d-inline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
/>,
                "className": "btn btn-negative text-white",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "tcon tcon-remove tcon-remove--chevron-right btn-negative text-white",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="pull-left server-name server-name--running"
            onClick={[Function]}
>
            test-server-1
</div>,
          <div
            className="pull-right"
>
            <div
                        className="d-inline"
            >
                        <CSSTransition
                                    classNames="server-item"
                                    in={true}
                                    timeout={
                                                Object {
                                                            "enter": 350,
                                                            "exit": 350,
                                                          }
                                    }
                                    unmountOnExit={true}
                        >
                                    <button
                                                className="btn btn-default mr-1 restart-button"
                                                onClick={[Function]}
                                    >
                                                <span
                                                            className="icon icon-arrows-ccw "
                                                />
                                    </button>
                        </CSSTransition>
            </div>
            <div
                        className="d-inline"
            >
                        <button
                                    className="btn btn-negative text-white"
                                    onClick={[Function]}
                                    type="button"
                        >
                                    <span
                                                className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
                                    />
                        </button>
            </div>
</div>,
        ],
        "className": "list-group-item server-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test-server-1",
            "className": "pull-left server-name server-name--running",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "test-server-1",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="d-inline"
>
                <CSSTransition
                                classNames="server-item"
                                in={true}
                                timeout={
                                                Object {
                                                                "enter": 350,
                                                                "exit": 350,
                                                              }
                                }
                                unmountOnExit={true}
                >
                                <button
                                                className="btn btn-default mr-1 restart-button"
                                                onClick={[Function]}
                                >
                                                <span
                                                                className="icon icon-arrows-ccw "
                                                />
                                </button>
                </CSSTransition>
</div>,
              <div
                className="d-inline"
>
                <button
                                className="btn btn-negative text-white"
                                onClick={[Function]}
                                type="button"
                >
                                <span
                                                className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
                                />
                </button>
</div>,
            ],
            "className": "pull-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CSSTransition
                  classNames="server-item"
                  in={true}
                  timeout={
                                    Object {
                                                      "enter": 350,
                                                      "exit": 350,
                                                    }
                  }
                  unmountOnExit={true}
>
                  <button
                                    className="btn btn-default mr-1 restart-button"
                                    onClick={[Function]}
                  >
                                    <span
                                                      className="icon icon-arrows-ccw "
                                    />
                  </button>
</CSSTransition>,
                "className": "d-inline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="btn btn-default mr-1 restart-button"
                    onClick={[Function]}
>
                    <span
                                        className="icon icon-arrows-ccw "
                    />
</button>,
                  "classNames": "server-item",
                  "in": true,
                  "timeout": Object {
                    "enter": 350,
                    "exit": 350,
                  },
                  "unmountOnExit": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span
                      className="icon icon-arrows-ccw "
/>,
                    "className": "btn btn-default mr-1 restart-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon icon-arrows-ccw ",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-negative text-white"
                  onClick={[Function]}
                  type="button"
>
                  <span
                                    className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
                  />
</button>,
                "className": "d-inline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="tcon tcon-remove tcon-remove--chevron-right btn-negative text-white"
/>,
                  "className": "btn btn-negative text-white",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "tcon tcon-remove tcon-remove--chevron-right btn-negative text-white",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ServerItem matches the snapshot when the server is stopped 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServerItem
    server={
        Object {
            "id": "test-server-2",
            "status": "stopped",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="pull-left server-name server-name--stopped"
          onClick={[Function]}
>
          test-server-2
</div>,
        <div
          className="pull-right"
>
          <div
                    className="d-inline"
          >
                    <CSSTransition
                              classNames="server-item"
                              in={false}
                              timeout={
                                        Object {
                                                  "enter": 350,
                                                  "exit": 350,
                                                }
                              }
                              unmountOnExit={true}
                    >
                              <button
                                        className="btn btn-default mr-1 restart-button"
                                        onClick={[Function]}
                              >
                                        <span
                                                  className="icon icon-arrows-ccw "
                                        />
                              </button>
                    </CSSTransition>
          </div>
          <div
                    className="d-inline"
          >
                    <button
                              className="btn tcon-transform btn-default"
                              onClick={[Function]}
                              type="button"
                    >
                              <span
                                        className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
                              />
                    </button>
          </div>
</div>,
      ],
      "className": "list-group-item server-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test-server-2",
          "className": "pull-left server-name server-name--stopped",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "test-server-2",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="d-inline"
>
              <CSSTransition
                            classNames="server-item"
                            in={false}
                            timeout={
                                          Object {
                                                        "enter": 350,
                                                        "exit": 350,
                                                      }
                            }
                            unmountOnExit={true}
              >
                            <button
                                          className="btn btn-default mr-1 restart-button"
                                          onClick={[Function]}
                            >
                                          <span
                                                        className="icon icon-arrows-ccw "
                                          />
                            </button>
              </CSSTransition>
</div>,
            <div
              className="d-inline"
>
              <button
                            className="btn tcon-transform btn-default"
                            onClick={[Function]}
                            type="button"
              >
                            <span
                                          className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
                            />
              </button>
</div>,
          ],
          "className": "pull-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CSSTransition
                classNames="server-item"
                in={false}
                timeout={
                                Object {
                                                "enter": 350,
                                                "exit": 350,
                                              }
                }
                unmountOnExit={true}
>
                <button
                                className="btn btn-default mr-1 restart-button"
                                onClick={[Function]}
                >
                                <span
                                                className="icon icon-arrows-ccw "
                                />
                </button>
</CSSTransition>,
              "className": "d-inline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn btn-default mr-1 restart-button"
                  onClick={[Function]}
>
                  <span
                                    className="icon icon-arrows-ccw "
                  />
</button>,
                "classNames": "server-item",
                "in": false,
                "timeout": Object {
                  "enter": 350,
                  "exit": 350,
                },
                "unmountOnExit": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="icon icon-arrows-ccw "
/>,
                  "className": "btn btn-default mr-1 restart-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon icon-arrows-ccw ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn tcon-transform btn-default"
                onClick={[Function]}
                type="button"
>
                <span
                                className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
                />
</button>,
              "className": "d-inline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
/>,
                "className": "btn tcon-transform btn-default",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="pull-left server-name server-name--stopped"
            onClick={[Function]}
>
            test-server-2
</div>,
          <div
            className="pull-right"
>
            <div
                        className="d-inline"
            >
                        <CSSTransition
                                    classNames="server-item"
                                    in={false}
                                    timeout={
                                                Object {
                                                            "enter": 350,
                                                            "exit": 350,
                                                          }
                                    }
                                    unmountOnExit={true}
                        >
                                    <button
                                                className="btn btn-default mr-1 restart-button"
                                                onClick={[Function]}
                                    >
                                                <span
                                                            className="icon icon-arrows-ccw "
                                                />
                                    </button>
                        </CSSTransition>
            </div>
            <div
                        className="d-inline"
            >
                        <button
                                    className="btn tcon-transform btn-default"
                                    onClick={[Function]}
                                    type="button"
                        >
                                    <span
                                                className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
                                    />
                        </button>
            </div>
</div>,
        ],
        "className": "list-group-item server-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test-server-2",
            "className": "pull-left server-name server-name--stopped",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "test-server-2",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="d-inline"
>
                <CSSTransition
                                classNames="server-item"
                                in={false}
                                timeout={
                                                Object {
                                                                "enter": 350,
                                                                "exit": 350,
                                                              }
                                }
                                unmountOnExit={true}
                >
                                <button
                                                className="btn btn-default mr-1 restart-button"
                                                onClick={[Function]}
                                >
                                                <span
                                                                className="icon icon-arrows-ccw "
                                                />
                                </button>
                </CSSTransition>
</div>,
              <div
                className="d-inline"
>
                <button
                                className="btn tcon-transform btn-default"
                                onClick={[Function]}
                                type="button"
                >
                                <span
                                                className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
                                />
                </button>
</div>,
            ],
            "className": "pull-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CSSTransition
                  classNames="server-item"
                  in={false}
                  timeout={
                                    Object {
                                                      "enter": 350,
                                                      "exit": 350,
                                                    }
                  }
                  unmountOnExit={true}
>
                  <button
                                    className="btn btn-default mr-1 restart-button"
                                    onClick={[Function]}
                  >
                                    <span
                                                      className="icon icon-arrows-ccw "
                                    />
                  </button>
</CSSTransition>,
                "className": "d-inline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="btn btn-default mr-1 restart-button"
                    onClick={[Function]}
>
                    <span
                                        className="icon icon-arrows-ccw "
                    />
</button>,
                  "classNames": "server-item",
                  "in": false,
                  "timeout": Object {
                    "enter": 350,
                    "exit": 350,
                  },
                  "unmountOnExit": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span
                      className="icon icon-arrows-ccw "
/>,
                    "className": "btn btn-default mr-1 restart-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon icon-arrows-ccw ",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn tcon-transform btn-default"
                  onClick={[Function]}
                  type="button"
>
                  <span
                                    className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
                  />
</button>,
                "className": "d-inline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default"
/>,
                  "className": "btn tcon-transform btn-default",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "tcon tcon-remove tcon-remove--chevron-right tcon-transform btn-default",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
